### PR TITLE
Feature 2 - To-do list screen and task creation, editing, deletion use cases/screens implemented

### PR DESCRIPTION
**Pull request by yyutiw**
A lot happened:
1. Condensed all of my creation interactors into one TaskCreationInteractor, changed format of all request/response models, input/output boundaries to reflect those changes. @celinelumqr I noticed you added some scheduling presenters to my old code, and threw them into my new interactor for you, but you can make changes if needed for your feature.
2. Combined editing and deleting into one screen and completed their implementations which hopefully should be functional
3. @TiareMar made a separate screen that displays the user's todo list, where each task is accompanied by an edit/delete button that triggers the use case mentioned above. Below is an image of that list with 3 Tasks:
image
5. @TiareMar also got serialization working for the TaskMap
6. Some other small changes: I moved around the code for initializing use cases and screens so that some screens would properly reset if returned to after visiting and then going back to main. Specifically, I moved task creation to the MainScreen class and placed the todo list screen there as well; I put all editing/deleting components within the todo list screen. I also changed the task IDs to be "taskname_username_coursename" instead of "datetime_username_coursename" after some discussion

What I'm working on now:
- [ ] implementing instructor screens -- will probably be done once everything is ready to be combined (separate branch)
- [ ] testing -- done on a different branch
- [x] adding javadoc comments throughout -- done on a different branch
- [x] task editing allows for titles to be changed now, but with the new task ID system this is problematic, so I will be disabling that text field
- [ ] fixing up code after @CC-3636 pushes her part, as I need to clean up some parts that drag a User through multiple layers
- [ ] fixing IntelliJ warnings
- [ ] the program needs a save and exit button

### CODE REVIEW

**Review by alyson647 — DISMISSED** [review]
Generally good adherence to clean architecture and SOLID principles. I found that there might be a violation of clean architecture in your screens because you import use cases and entities as well as in controllers, where I found entities imported. Documentation could be added but I know that you're working on that in another branch. Otherwise seems good.

**Review by celinelumqr — APPROVED** [review]
looks good to me, a lot of functionality implemented for your feature with this PR

**Review by TiareMar — DISMISSED** [review]
Approving because code functionality looks good! Documentation and more testing can be added later. Don't forget to pull in the latest from main before merging!!

**Review by celinelumqr — DISMISSED** [review]

File: src/main/java/Main.java
Comment (TiareMar) [reply src/main/java/Main.java:110]: If you check the other screens (ex. CalendarScreen), for their "back" or "return" buttons, they should be going back to "StudentMain" screen. You will probably also need to change your own screens so that any "back" buttons go back to "StudentMain"

**Review by TiareMar — APPROVED** [review]
Approving so that others can get your code from main but please fix these bugs as soon as you can.

**Review by TiareMar — APPROVED** [review]
Again, approving in the interest of getting your code on main but please see the small issues I mentioned

**Review by celinelumqr — APPROVED** [review]

**Review by jltng — APPROVED** [review]
Looks good! Minor issues that don't effect overall functionality of code

File: src/main/java/Main.java
Comment (yyutiw) [reply src/main/java/Main.java:41]: To address your first comment: I ended up making a small adjustment in the load method itself (in TaskMap) so that the map is initialized to a blank HashMap if reading from a file throws an exception. I did this because it was printing an error message before when the map was getting initialized to null.

Also, all of the files and refactoring were a complete mistake from trying to merge changes and push commits quickly! That's 100% my bad

File: src/main/java/Main.java
Comment (yyutiw) [reply src/main/java/Main.java:98]: Yes I can do that! I was just working on testing while my PR was getting approved, will remove all of the unnecessary user params in the near future

File: src/main/java/entities/TaskMap.java
Comment (yyutiw) [reply src/main/java/entities/TaskMap.java:7]: I'll maybe try to remove the save and load files from TaskMap and throw them in a new class, and then just use the setter to initialize the map

File: src/main/java/screens/task_management/task_creation_screens/ChooseTaskCreateScreen.java
Comment (yyutiw) [reply src/main/java/screens/task_management/task_creation_screens/ChooseTaskCreateScreen.java:3]: Might be able to resolve this with recent changes

File: src/main/java/screens/task_management/task_creation_screens/ChooseTaskCreateScreen.java
Comment (yyutiw) [reply src/main/java/screens/task_management/task_creation_screens/ChooseTaskCreateScreen.java:14]: so true I was just alt-entering all of the imports

File: src/main/java/screens/task_management/todolist_screens/ToDoListScreen.java
Comment (yyutiw) [reply src/main/java/screens/task_management/todolist_screens/ToDoListScreen.java:5]: Yeah my bad don't know why I didn't use the task ID in the displayers

**Pull request merged**
